src/util.js

```diff
@@ -1,12 +1,16 @@
-export function _normalizeAbsolutePath(path, hasPushState) {
+export function _normalizeAbsolutePath(path, hasPushState, absolute = false) {
   if (!hasPushState && path[0] !== '#') {
     path = '#' + path;
   }
 
+  if (hasPushState && absolute) {
+    path = path.substring(1, path.length);
+  }
+
   return path;
 }
 
-export function _createRootedPath(fragment, baseUrl, hasPushState) {
+export function _createRootedPath(fragment, baseUrl, hasPushState, absolute) {
```

Comment:
Perhaps you're right about that - the separation of concerns does make sense - but that would then mean that relative routes would still need to add a hash, or generate the full URL and then extract the desired section. It also means that generating a link to an empty route ('') would always have the hash included. Right now, the default generate function creates a relative path. Do we intend to change that?